File: BrokerCommon/swimmer.cs
using System;

namespace BrokerCommon
{
    public class Swimmer
    {
        public SocketLayer Client { get; }
        public string Id { get; set; }

        public Swimmer(SocketLayer client, string swimmerId)
        {
            this.Client = client;
            this.Id = swimmerId;
        }


        public void SendMessage(Query query)
        {
            query.Add("~ToSwimmer~", this.Id);
            Client.SendMessage(query);
        }

        public void SendMessageWithResponse(Query query, Action<Query> callback) 
        {
            query.Add("~ToSwimmer~", this.Id);
            Client.SendMessageWithResponse(query, callback);
        }
    }
}

File: OnPoolClientTS/src/swimmer.ts
import {Query} from "./common/query";
import {SocketLayer} from "./common/socketLayer";

export class Swimmer {
    private Client:SocketLayer;
    public Id: string;
    constructor(clientPool: SocketLayer, swimmerId:string) {
        this.Client = clientPool;
        this.Id = swimmerId;
    }
    public SendMessage(query: Query): void {
        query.Add("~ToSwimmer~", this.Id);
        this.Client.SendMessage(query);
    }
    public SendMessageWithResponse(query: Query, callback: (response: Query) => void): void {
        query.Add("~ToSwimmer~", this.Id);
        this.Client.SendMessageWithResponse(query, (response) => {
            callback(response);
        });
    }
}